src/components/site/debug/index.jsx

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable no-bitwise */
+import React from 'reactn';
+import { Trans } from 'gatsby-plugin-react-i18next';
+import ls from 'local-storage';
+
+class Debug extends React.PureComponent {
+  getDownloadLink = () => {
+    const savedState = JSON.stringify(ls(this.global.investigation));
+    const savedStateHash = this.hashKeys(Object.keys(savedState));
+    const state = JSON.stringify(this.global);
+    const stateHash = this.hashKeys(Object.keys(state));
+    const browser = navigator.userAgent;
+    const location = window.location.href;
+    const online = navigator.onLine;
+    const date = new Date();
+    const dimensions = `${window.innerWidth}px x ${window.innerHeight}px`;
+
+    const data = new Blob(
+      [
+        `Location: ${location}\nDate: ${date}\nBrowser: ${browser}\nOnline: ${online}\nDimensions: ${dimensions}\n\nLocal storage - ${savedStateHash}\n${savedState}\n\nState - ${stateHash}\n${state}`,
+      ],
+      { type: 'text/plain' }
+    );
+    return window.URL.createObjectURL(data);
+  };
+
+  getFileName = () => `debug-${new Date().getTime()}.txt`;
+
+  hashKeys = keys =>
```

Comment:
simple hash for the keys in local storage and the application state to generate a rudimentary versioning system based on the number of keys and their name.